### PR TITLE
Rnacode

### PR DESCRIPTION
**Pull request by smithcr**
mostly done, a couple of issues
still needs a .shed.yml file
with planemo test:
```
All 2 test(s) executed passed.
rbc_rnacode[0]: failed
rbc_rnacode[1]: passed
```
when create EPS plots is selected, the output EPS files appear in a dataset AND separately in history, perhaps @TorHou can suggest?
galaxy does not natively display/view the generated eps files

### CODE REVIEW

**Comment by bgruening** [comment]
@smithcr for visualisation please install the visualisation from: https://github.com/bgruening/galaxytools/tree/master/visualisations

File: tools/rna_tools/rnacode/rnacode.xml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        
+        #if $cond_generateEPS.generateEPS == 'create'
+            --eps
+            #if $cond_generateEPS.eps_cutoff and $cond_generateEPS.eps_cutoff is not None
+                --eps-cutoff $cond_generateEPS.eps_cutoff
+            #end if
+        #end if
+        
+        #if $cond_scoringParameters.scoringParameters == 'custom'
+            --pars $pars
```
Comment (bgruening) [inline tools/rna_tools/rnacode/rnacode.xml]: Please put text parameters in quotes.

File: tools/rna_tools/rnacode/rnacode.xml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        <exit_code range="1:"  level="fatal" description="Error occurred. Please check Tool Standard Error"/>
+        <exit_code range=":-1" level="fatal" description="Error occurred. Please check Tool Standard Error"/>
+    </stdio>
+    <version_command>RNAcode --version</version_command>
+    <command>
+    <![CDATA[
+        RNAcode
+        
+        #if $outputFormat.value == 'default'
+            --tabular
```
Comment (bgruening) [inline tools/rna_tools/rnacode/rnacode.xml]: If you name it like `--tabular` and `--gtf` you can get rid of the #if

File: tools/rna_tools/rnacode/rnacode.xml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+            <filter>cond_generateEPS['generateEPS'] == "create"</filter>
+            <discover_datasets pattern="(?P&lt;designation&gt;.*)\.eps" directory="eps" ext="eps" visible="true"/>
+        </collection>
+    </outputs>
+    <tests>
+        <test>
+            <param name="alignment" value="coding.aln"/>
+            <param name="generateEPS" value="nocreate"/>
+            <param name="outputFormat" value="default"/>
+            <output name="outFileDefault" ftype="tabular" file="rnacode_result1.tabular" compare="sim_size"/>
```
Comment (bgruening) [inline tools/rna_tools/rnacode/rnacode.xml]: Please insert a comment, why `sim_size` is needed.

File: tools/rna_tools/rnacode/rnacode.xml [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+            with 1.
+
+6. Name      
+            The name of the reference sequence as given in the input alignment.
+  
+7. Start     
+            The nucleotide position in the reference sequence of the
+
+8. End       
+            predicted coding region. If no genomic coordinates are given
```
Comment (bgruening) [inline tools/rna_tools/rnacode/rnacode.xml]: Here is something wrong!

File: tools/rna_tools/rnacode/tool_dependencies.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
+<tool_dependency>
+    <package name="rnacode" version="0.3">
+        <install version="1.0">
+            <actions>
+                <action type="download_by_url">https://raw.githubusercontent.com/bgruening/download_store/master/RNAcode/RNAcode-0.3.tar.gz</action>
+                <action type="shell_command">./configure --prefix=$INSTALL_DIR</action>
+                <action type="shell_command">make</action>
```
Comment (bgruening) [inline tools/rna_tools/rnacode/tool_dependencies.xml]: `<action type="autoconf"/>` will do exactly this.

File: tools/rna_tools/rnacode/tool_dependencies.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+                <action type="shell_command">make install</action>
+                <action type="set_environment">
+                    <environment_variable name="PATH" action="prepend_to">$INSTALL_DIR/bin</environment_variable>
+                    <environment_variable name="RNACODE_ROOT_PATH">$INSTALL_DIR</environment_variable>
```
Comment (bgruening) [inline tools/rna_tools/rnacode/tool_dependencies.xml]: action is missing :)

**Comment by smithcr** [comment]
only issue still remaining is the failed/but not failed planemo test....

fixes:
    dataset collection problem fixed by setting visible=false
    unviewability of eps files is a browser problem

**Pull request merged**